tests/integration/models/test_eval_methods_mass_balance.py

```diff
@@ -0,0 +1,420 @@
+"""Provides tests for AD methods used in the single phase flow model.
+
+We provide two tests:
+
+    (1) `test_tested_vs_testable_methods_single_phase_flow`: This test checks that
+      all the testable methods (see docstring of `setup_utils.get_testable_methods()`
+      to see what constitutes a testable method) of the single-phase flow model are
+      included in the parametrization of `test_ad_operator_methods_single_phase_flow`,
+      and,
+
+    (2) `test_ad_operator_methods_single_phase_flow`: A parametrized test where each
+      tested method is evaluated and their output is compared to expected values.
+
+"""
+
+from __future__ import annotations
+
+from typing import Callable
+
+import numpy as np
+import pytest
+import scipy.sparse as sps
+
+import porepy as pp
+from porepy.applications.md_grids.model_geometries import (
+    SquareDomainOrthogonalFractures,
+)
+from porepy.models.fluid_mass_balance import SinglePhaseFlow
+from tests.integration.models import setup_utils
+
+
+@pytest.fixture(scope="function")
+def model_setup():
+    """Minimal compressible single-phase flow setup with two intersecting fractures.
+
+    The model is set up with realistic physical parameters using water and granite.
+    A 2x2 Cartesian grid on a unit domain is employed. One horizontal and one vertical
+    fractures are included in the domain. This results in the following `mdg`:
+
+        Maximum dimension present: 2
+        Minimum dimension present: 0
+        1 grids of dimension 2 with in total 4 cells
+        2 grids of dimension 1 with in total 4 cells
+        1 grids of dimension 0 with in total 1 cells
+        2 interfaces between grids of dimension 2 and 1 with in total 8 mortar cells.
+        2 interfaces between grids of dimension 1 and 0 with in total 4 mortar cells.
+
+    Non-zero (but constant) states for the primary variables (pressure and interface
+    fluxes) are assigned to the `iterate` states. Note that the model is not
+    effectively run, but it is discretized via `prepare_simulation()`.
+
+    Returns:
+        Prepared-for-simulation model with non-trivial primary variables.
+
+    """
+
+    class Model(SquareDomainOrthogonalFractures, SinglePhaseFlow):
+        """Single phase flow model in a domain with two intersecting fractures."""
+
+    # Material constants
+    solid = pp.SolidConstants(setup_utils.granite_values)
+    fluid = pp.FluidConstants(setup_utils.water_values)
+
+    # Declare model parameters
+    params = {
+        "material_constants": {"solid": solid, "fluid": fluid},
+        "fracture_indices": [0, 1],
+        "grid_type": "cartesian",
+        "meshing_arguments": {"cell_size_x": 0.5, "cell_size_y": 0.5},
+    }
+
+    # Instantiate the model setup
+    setup = Model(params)
+
+    # Prepare to simulate
+    setup.prepare_simulation()
+    setup.set_well_network()
+
+    # Set constant but non-zero values for the primary variables
+    if hasattr(setup, "pressure_variable"):
```

Comment:
Is this if needed?

Reply:
Not really. Now the primary variables should be constrained by each model. I will remove them.